src/main/kotlin/org/rust/ide/wordSelection/RsGroupSelectionHandler.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.wordSelection
+
+import com.intellij.codeInsight.editorActions.ExtendWordSelectionHandlerBase
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.util.TextRange
+import com.intellij.openapi.util.text.LineTokenizer
+import com.intellij.psi.PsiComment
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiWhiteSpace
+import org.rust.lang.core.psi.RsElementTypes
+import org.rust.lang.core.psi.RsMembers
+import org.rust.lang.core.psi.RsStmt
+import java.util.*
+
+class RsGroupSelectionHandler : ExtendWordSelectionHandlerBase() {
+    override fun canSelect(e: PsiElement): Boolean
+            = e is RsStmt || RsFieldSelectionHandler.isFieldDecl(e) || e.parent is RsMembers || e is PsiComment
+
+
+    override fun select(e: PsiElement, editorText: CharSequence, cursorOffset: Int, editor: Editor): List<TextRange>? {
+        // see com.intellij.codeInsight.editorActions.wordSelection.StatementGroupSelectioner for the reference implementation
+
+        val result = ArrayList<TextRange>()
+
+        var startElement = e
+        var endElement = e
+
+        while (startElement.prevSibling != null) {
```

Comment:
Consider use sequences here too. But in such complex control flow it may be ok

Reply:
I'll better leave it as is - it's easier to compare with the original code